src/preamble.js

```diff
@@ -1237,6 +1238,15 @@ if (typeof Atomics === 'undefined') {
   Atomics['xor'] = function(t, i, v) { var w = t[i]; t[i] ^= v; return w; }
 }
 
+#else
+// TODO: Something smart about BINARYEN_MEM_MAX. Some default, also rename it.
```

Comment:
is it right that wasm can't support memory growth with shared memory? if so, we can just show an error if growth or max > initial?

Reply:
It can support growth, just not unconstrained growth. Or in other words, you can have max > initial, but you can't set an initial without a max.

Reply:
I see, ok.

I still don't understand this code, though. Why do we create the memory here instead of later down where we create it normally (which also handles MEM_MAX)?

Reply:
Basically just because of the overall structure of the ifdefs. All the pthreads stuff is first, and everything below is in the else (not pthreads). This whole nest of code is really ugly and complex because of all the options, should probably cleaned up and/or the options rethought.

Reply:
Need to resolve this comment - I think we can make command line requirement to pass `BINARYEN_MEM_MAX` so that frontend makes sure in multithreaded mode this gets set?

Reply:
OK, made it a requirement to add `WASM_MEM_MAX` when using wasm+pthreads and memory growth. I tried to add a test but it doesn't look like memory growth on a thread is working right now. Will address that separately.